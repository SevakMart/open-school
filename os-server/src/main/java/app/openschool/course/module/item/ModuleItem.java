package app.openschool.course.module.item;

import app.openschool.course.module.Module;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name = "module_item")
public class ModuleItem {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(nullable = false)
  private Long id;

  @Column(name = "module_item_type", nullable = false)
  private String moduleItemType;

  @Column(name = "estimated_time")
  private Long estimatedTime;

  @ManyToOne
  @JoinColumn(name = "module_id")
  private Module module;

  public ModuleItem() {}

  public ModuleItem(Long id, String moduleItemType, Long estimatedTime, Module module) {
    this.id = id;
    this.moduleItemType = moduleItemType;
    this.estimatedTime = estimatedTime;
    this.module = module;
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getModuleItemType() {
    return moduleItemType;
  }

  public void setModuleItemType(String moduleItemType) {
    this.moduleItemType = moduleItemType;
  }

  public Long getEstimatedTime() {
    return estimatedTime;
  }

  public void setEstimatedTime(Long estimatedTime) {
    this.estimatedTime = estimatedTime;
  }

  public Module getModule() {
    return module;
  }

  public void setModule(Module module) {
    this.module = module;
  }
}
