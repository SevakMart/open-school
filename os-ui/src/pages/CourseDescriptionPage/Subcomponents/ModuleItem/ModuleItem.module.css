.mainContainer {
  background-color: #F9F9FD;
  border-radius: 8px;
  margin-right: 10px;
  margin-left: 10px;
}
.mainContainer .mainContent {
  margin: 10px 8px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}
.mainContainer .mainContent .titleAndEllipseIcon {
  display: inline-block;
}
.mainContainer .mainContent .titleAndEllipseIcon .moduleTitle {
  display: inline;
  margin-left: 16px;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #333941;
}
.mainContainer .mainContent .chevronIsOpen {
  cursor: pointer;
  transform: rotate(90deg);
  transition: 0.5s;
}
.mainContainer .mainContent .chevronIsClosed {
  cursor: pointer;
  transform: rotate(0deg);
  transition: 0.5s;
}
.mainContainer .moduleDescriptionIsOpen {
  margin: 24px 16px;
  text-align: justify;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  color: #5E617B p;
  color-margin: 0 0 0 8px;
}
.mainContainer .moduleDescriptionIsOpen .moduleItemLink {
  -webkit-appearance: none;
     -moz-appearance: none;
          appearance: none;
  background-color: transparent;
  border: none;
  color: #5E617B;
  text-decoration: underline;
  cursor: pointer;
  margin: 0;
  padding: 0;
  font: inherit;
}
.mainContainer .moduleDescriptionIsClosed {
  display: none;
}/*# sourceMappingURL=ModuleItem.module.css.map */