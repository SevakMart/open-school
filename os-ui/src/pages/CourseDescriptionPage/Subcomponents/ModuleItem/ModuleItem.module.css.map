{"version":3,"sources":["ModuleItem.module.scss","ModuleItem.module.css","../../../../styles/Mixins/FlexMixins.scss","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss"],"names":[],"mappings":"AAKA;EACE,yBAAA;EACA,kBAAA;EACA,kBAAA;EACA,iBAAA;ACJF;ADME;EACE,gBAAA;EEXF,aAAA;EACA,mBFWqB;EEVrB,mBFU0B;EET1B,8BFSkC;ACDpC;ADGI;EACE,qBAAA;ACDN;ADGM;EACE,eAAA;EACA,iBAAA;EGnBN,yBCDY;EDEZ,kBHmBwC;EGlBxC,gBHkBgD;EGjBhD,eHiBqD;EGhBrD,iBHgB2D;EACrD,cKjBW;AJoBnB;ADCI;EACE,eAAA;EACA,wBAAA;EACA,gBAAA;ACCN;ADEI;EACE,eAAA;EACA,uBAAA;EACA,gBAAA;ACAN;ADIE;EACE,iBAAA;EACA,mBAAA;EGxCF,yBCDY;EDEZ,kBHwCoC;EGvCpC,gBHuC4C;EGtC5C,eHsCiD;EGrCjD,iBHqCuD;EACrD,gBAAA;EAGI,uBAAA;ACAR;ADGM;EACJ,wBAAA;KAAA,qBAAA;UAAA,gBAAA;EACA,6BAAA;EACA,YAAA;EACA,cKnDgB;ELoDhB,0BAAA;EACA,eAAA;EACA,SAAA;EACA,UAAA;EACA,aAAA;ACDF;ADME;EACE,aAAA;ACJJ","file":"ModuleItem.module.css"}