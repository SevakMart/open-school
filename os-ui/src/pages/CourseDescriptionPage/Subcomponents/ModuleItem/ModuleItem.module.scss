@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";
@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";

.mainContainer,.enrolledMainContainer {
  background-color: $course-module-backgraund-color;
  border-radius: 8px;
  display: flex;
  flex-wrap: wrap;
  padding: 0px 16px;
  position: relative;

  .mainContent {
    @include flexStyle(row, center, space-between);

    .titleAndEllipseIcon {
      display: flex;
      align-items: center;

      .titleAndIconWrapper {
        display: flex;
        align-items: center;

        .ellipseIcon {
          display: flex;
          align-items: center;
          position: relative;
        }

        .vector {
          position: relative;
          right: 9px;
        }
      }

      .moduleTitle {
        @include fontStyle($primary-font, normal, 600, 20px, 24px);
        color: $black-text-color;
        max-width: 650px;
        margin-left: 8px;
      }

      .remainingTimeContainer {
        display: flex;
        align-items: center;
      }

      .remainingTime {
        position: relative;
        width: 24px;
        height: 24px;
        border-radius: 50%;
        box-shadow: 0px 2px 35px rgba(217, 219, 233, 0.24);
        margin-left: 30px;
      }

      .remainingTimeContent {
        display: inline-block;
        margin-left: 8px;
        margin-right: 78px;
      }
    }

	.chevronIsOpen,
    .chevronIsClosed {
	  position: absolute;
	  right: 16px;
      cursor: pointer;
      transform: rotate(0deg);
      transition: transform 0.5s;
     

      &.chevronIsOpen {
        transform: rotate(90deg);
      }
    }
  }

  .moduleDescriptionIsOpen {
    margin-top: 24px;
	margin-bottom: 24px;
    padding: 10px;
    text-align: justify;
    @include fontStyle($primary-font, normal, 400, 16px, 24px);
    color: $main-text-color;

    .moduleTitle {
      flex-basis: 100%; 
    }

    .moduleItemLink {
      appearance: none;
      background-color: transparent;
      border: none;
      color: $main-text-color;
      text-decoration: underline;
      cursor: pointer;
      margin: 0;
      padding: 0;
      font: inherit;
    }
  }

  .moduleDescriptionIsClosed {
    display: none;
  }
}

.enrolledMainContainer{
	width: 564px;
}

.mainContainer{
	width: 100%;
}