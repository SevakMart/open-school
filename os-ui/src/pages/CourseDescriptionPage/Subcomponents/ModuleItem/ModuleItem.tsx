import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import ArrowRightIcon from '../../../../assets/svg/ArrowRight.svg';
import EllipseIcon from '../../../../assets/svg/Ellipse.svg';
import { CourseDescriptionType } from '../../../../types/CourseTypes';
import styles from './ModuleItem.module.scss';

type Module=CourseDescriptionType['modules'][number]
const ModuleItem = ({ moduleInfo }:{moduleInfo:Module}) => {
  const { t } = useTranslation();
  const [moduleListIsOpen, setModuleListIsOpen] = useState(false);
  const {
    mainContainer, mainContent, ellipseIcon, titleAndEllipseIcon, moduleTitle,
    chevronIsOpen, chevronIsClosed, moduleDescriptionIsOpen, moduleDescriptionIsClosed,
  } = styles;
  const openModuleList = () => {
    setModuleListIsOpen((prevState) => !prevState);
  };

  return (
    <div className={mainContainer}>
      <div className={mainContent}>
        <div className={titleAndEllipseIcon}>
          <img className={ellipseIcon} src={EllipseIcon} alt="ellipse icon" />
          <p className={moduleTitle}>{t(moduleInfo.title)}</p>
        </div>
        <img className={moduleListIsOpen ? chevronIsOpen : chevronIsClosed} src={ArrowRightIcon} alt="chevron" onClick={openModuleList} />
      </div>
      <div className={moduleListIsOpen ? moduleDescriptionIsOpen : moduleDescriptionIsClosed}>
        <p>{t(moduleInfo.description)}</p>
        <button
          type="button"
          style={{
            backgroundColor: 'transparent',
            border: 'none',
            color: 'blue',
            textDecoration: 'underline',
            cursor: 'pointer',
          }}
        >
          {moduleInfo.link}
        </button>
      </div>
    </div>
  );
};
export default ModuleItem;
