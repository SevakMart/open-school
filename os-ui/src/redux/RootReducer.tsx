import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
