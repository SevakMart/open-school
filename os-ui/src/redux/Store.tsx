import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
