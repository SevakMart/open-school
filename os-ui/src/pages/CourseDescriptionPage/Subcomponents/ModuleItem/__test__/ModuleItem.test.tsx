import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { PayloadAction } from '@reduxjs/toolkit';
import { store } from '../../../../../redux/Store';
import ModuleItem from '../ModuleItem';

const moduleInfo = {
  title: 'React js',
  description: 'It talks about React',
  moduleItemSet: [{ title: 'Redux' }],
};
jest.mock('redux-state-sync', () => ({
  createStateSyncMiddleware:
    () => () => (next: (action: PayloadAction) => void) => (action: PayloadAction) => next(action),
  initMessageListener: () => jest.fn(),
}));

describe('Create test cases for ModuleItem component', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ModuleItem moduleInfo={moduleInfo} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
