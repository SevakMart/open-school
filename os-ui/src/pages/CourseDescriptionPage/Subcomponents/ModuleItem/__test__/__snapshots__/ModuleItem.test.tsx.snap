// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for ModuleItem component Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="mainContent"
    >
      <div
        class="titleAndEllipseIcon"
      >
        <img
          alt="ellipse icon"
          class="ellipseIcon"
          src="Ellipse.svg"
        />
        <p
          class="moduleTitle"
        >
          React js
        </p>
      </div>
      <img
        alt="chevron"
        class="chevronIsClosed"
        src="ArrowRight.svg"
      />
    </div>
    <div
      class="moduleDescriptionIsClosed"
    >
      <p>
        It talks about React
      </p>
      <button
        style="background-color: transparent; color: blue; text-decoration: underline; cursor: pointer;"
        type="button"
      />
    </div>
  </div>
</DocumentFragment>
`;
